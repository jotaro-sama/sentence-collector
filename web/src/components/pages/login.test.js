import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';

import ConnectedLogin from '../../containers/login';
import { getStore } from '../../tests/testUtils';

test('Login button is disabled without input values', () => {
  const initialState = {
    app: {
      authed: false,
      username: null,
      errormessage: null,
      loginDisabled: true,
    },
  };
  let store = getStore(initialState);
  let component = mount(<Provider store={store}><ConnectedLogin /></Provider>);

  expect(component).toMatchSnapshot();
  expect(component.find('form#login button').is('[disabled]')).toBeTruthy();
});

