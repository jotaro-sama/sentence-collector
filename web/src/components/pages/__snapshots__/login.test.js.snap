// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login button is disabled without input values 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Login)>
    <Login
      authed={false}
      checkLoginInput={[Function]}
      login={[Function]}
      loginDisabled={true}
      logout={[Function]}
      username={null}
    >
      <form
        id="login"
        onSubmit={[Function]}
      >
        <h2>
          Log in
        </h2>
        <p>
          Choose your username and password to sign up. If you already have a username, please sign in with the existing password. This login is separate from the main Common Voice site.
        </p>
        <section
          className="form-message"
        />
        <section>
          <label
            htmlFor="username"
          >
            username (alphanumeric, no email)
          </label>
          <input
            id="username"
            onChange={[Function]}
            type="text"
          />
          <label
            htmlFor="password"
          >
            password
          </label>
          <input
            id="password"
            onChange={[Function]}
            type="password"
          />
          <button
            disabled={true}
          >
            Submit
          </button>
        </section>
      </form>
    </Login>
  </Connect(Login)>
</Provider>
`;
